From 1c71d74635c2b39a15c449e75eacae23b3d4f1b8 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 30 May 2019 08:30:06 +0800
Subject: [PATCH] policy/modules/services/rpc: make rpcd_t MLS trusted for
 reading from files up to its clearance

Fixes:
type=AVC msg=audit(1559176077.169:242): avc:  denied  { search } for
pid=374 comm="rpc.statd" name="journal" dev="tmpfs" ino=9854
scontext=system_u:system_r:rpcd_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 tclass=dir
permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/rpc.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/rpc.te b/policy/modules/services/rpc.te
index 7c0b37ddc..ef6cb9b63 100644
--- a/policy/modules/services/rpc.te
+++ b/policy/modules/services/rpc.te
@@ -185,6 +185,8 @@ seutil_dontaudit_search_config(rpcd_t)
 
 userdom_signal_all_users(rpcd_t)
 
+mls_file_read_to_clearance(rpcd_t)
+
 ifdef(`distro_debian',`
 	term_dontaudit_use_unallocated_ttys(rpcd_t)
 ')
-- 
2.17.1

